OCPP-J/src/main/java/eu/chargetime/ocpp/JSONCommunicator.java

```diff
@@ -77,38 +77,37 @@ public JSONCommunicator(Radio radio) {
     super(radio);
   }
 
-  private class ZonedDateTimeSerializer implements JsonSerializer<ZonedDateTime> {
+  private static class ZonedDateTimeSerializer implements JsonSerializer<ZonedDateTime>, JsonDeserializer<ZonedDateTime> {
 
     @Override
     public JsonElement serialize(
         ZonedDateTime zonedDateTime, Type type, JsonSerializationContext jsonSerializationContext) {
       return new JsonPrimitive(zonedDateTime.format(DateTimeFormatter.ISO_INSTANT));
     }
-  }
-
-  public class ZonedDateTimeDeserializer implements JsonDeserializer<ZonedDateTime> {
 
     @Override
     public ZonedDateTime deserialize(
-        JsonElement jsonElement, Type type, JsonDeserializationContext jsonDeserializationContext)
-        throws JsonParseException {
+            JsonElement jsonElement, Type type, JsonDeserializationContext jsonDeserializationContext)
+            throws JsonParseException {
       return ZonedDateTime.parse(jsonElement.getAsJsonPrimitive().getAsString());
     }
   }
 
+  private static Gson gson;
+
+  static {
+    GsonBuilder builder = new GsonBuilder();
+    builder.registerTypeAdapter(ZonedDateTime.class, new ZonedDateTimeSerializer());
+    gson = builder.create();
+  }
```

Comment:
Never seen this before, I had to look it up. It's called static block and it runs while the class loader runs, before main is called. Interesting.